_posts/2023-06-27-redis.md

```diff
@@ -0,0 +1,23 @@
+
+The second more durable persistence mode is Append Only File which was introduced in Redis 1.1 to solve the drawbacks of snapshotting. In this mode, every redis command, is appended to a file as a log. Very much like WAL logs for conventional RDBMSs. This way, you can build the entire database  by replaying the entire file. If can be argued that sequential writes to file are significantly faster than random access but is is still significantly slower than writing in memory, and this goes against the essence of what redis is, which is an in-memory data store. If you are going to be writing each command to a file, why not use a datastore that is designed for that in the first place. If you use AOF it means redis is going to call fsync at a point in time which can be configured in 1 of 3 ways. `appendfsync always` will call fsync on each command. This is very very safe. But with this option you might as well through redis, out the window, because the performance becomes infirior to every database designed to fsync on write. if you use this mode without a [Battery Backed Write Cache (BBWC) RAID controller](https://serverfault.com/questions/65096/battery-backed-write-cache), you will get fucked. Have fun figuring out what went wrong. `appendfsync everysec` will call `fsync` every second which means at most you loose a second of data. This might sound reasonable but is not without its drawbacks. if you have a update intensive application like a counter that updates many times in an second, you end up with a needlessly huge AOF file when in essence the data payload might only have a small foot print. that said is is an easier choice and has been the new default since Redis 2.4. `appendfsync no` delegates the calling of `fsync` to the operating system. This is the fastest and the least safe method amongst all the `appendfsync` options. Normally Linux will flush data every 30 seconds with this configuration, but it's up to the kernel's exact tuning. 
```

Comment:
```suggestion
The second more durable persistence mode is Append Only File, introduced in Redis 1.1 to solve the drawbacks of snapshotting. In this mode, every Redis command is appended to a file as a log. Very much like WAL logs for conventional RDBMSs. This way, you can build the entire database by replaying the entire file. It can be argued that sequential writes to files are significantly faster than random access but is is still significantly slower than writing in memory, and this goes against the essence of what Redis is, which is an in-memory data store. If you are going to be writing each command to a file, why not use a datastore that is designed for that in the first place? If you use AOF, it means Redis is going to call fsync at a point in time which can be configured in 1 of 3 ways. `appendfsync always` will call fsync on each command. This is very very safe. But with this option, you might as well throw Redis out the window because the performance becomes inferior to every database designed to fsync on write. If you use this mode without a [Battery Backed Write Cache (BBWC) RAID controller](https://serverfault.com/questions/65096/battery-backed-write-cache), you will get fucked. Have fun figuring out what went wrong. `appendfsync everysec` will call `fsync` every second, which means, at most you lose a second of data. This might sound reasonable, but it is not without its drawbacks. If you have an update-intensive application like a counter that updates many times in a second, you end up with a needlessly huge AOF file when in essence, the data payload might only have a small footprint. That said, it is an easier choice and has been the new default since Redis 2.4. `appendfsync no` delegates the calling of `fsync` to the operating system. This is the fastest and the least safe method amongst all the `appendfsync` options. Normally Linux will flush data every 30 seconds with this configuration, but it's up to the kernel's exact tuning. 
```